managing_providers/_topics/embedded_workflows.md

```diff
@@ -0,0 +1,433 @@
+
+   Within the workflow code, you need to specify the states that your workflow requires, including any next steps. For each of the steps in the workflow, a docker container is called. The container defines what happens for that state. For example, a docker container can run to clone a template.
+
+2. Build the docker containers that are required for the workflow.
+
+#### Example: Provisioning Workflow
+
+```asl
+{
+  "Comment": "Provision a VMware VM.",
+  "StartAt": "CloneTemplate",
+  "States": {
+    "CloneTemplate": {
+      "Type": "Task",
+      "Resource": "docker://docker.io/agrare/clone-template:latest",
+      "Next": "CheckTaskComplete",
+      "Credentials": {
+        "api_user.$": "$.api_user",
+        "api_password.$": "$.api_password",
+        "vcenter_user.$": "$.vcenter_user",
+        "vcenter_password.$": "$.vcenter_password"
+      },
+      "Parameters": {
+        "PROVIDER_ID.$": "$.dialog_provider",
+        "VERIFY_SSL": false,
+        "TEMPLATE.$": "$.dialog_source_template",
+        "NAME.$": "$.dialog_vm_name"
+      }
+    },
+
+    "CheckTaskComplete": {
+      "Type": "Task",
+      "Resource": "docker://docker.io/agrare/check-task-complete:latest",
+      "Next": "PollTaskComplete",
+      "Credentials": {
+        "vcenter_user.$": "$.vcenter_user",
+        "vcenter_password.$": "$.vcenter_password"
+      },
+      "Parameters": {
+        "VCENTER_HOST.$": "$.vcenter_host",
+        "TASK.$": "$.task"
+      }
+    },
+
+    "PollTaskComplete": {
+      "Type": "Choice",
+      "Choices": [
+        {
+          "Variable": "$.state",
+          "StringEquals": "success",
+          "Next": "PowerOnVM"
+        },
+        {
+          "Variable": "$.state",
+          "StringEquals": "running",
+          "Next": "RetryState"
+        },
+        {
+          "Variable": "$.state",
+          "StringEquals": "error",
+          "Next": "FailState"
+        }
+      ],
+      "Default": "FailState"
+    },
+
+    "RetryState": {
+      "Type": "Wait",
+      "Seconds": 5,
+      "Next": "CheckTaskComplete"
+    },
+
+    "PowerOnVM": {
+      "Type": "Task",
+      "Resource": "docker://docker.io/agrare/power-on-vm:latest",
+      "Next": "SuccessState",
+      "Credentials": {
+        "vcenter_user.$": "$.vcenter_user",
+        "vcenter_password.$": "$.vcenter_password"
+      },
+      "Parameters": {
+        "VCENTER_HOST.$": "$.vcenter_host",
+        "VM.$": "$.vm"
+      }
+    },
+
+    "FailState": {
+      "Type": "Fail",
+      "Error": "FailStateError",
+      "Cause": "No Matches!"
+    },
+
+    "SuccessState": {
+      "Type": "Succeed"
+    }
+  }
+}
+```
+
+### Viewing workflow details
+
+When workflows are available, you can view the details for each workflow, such as to verify the payload. To view the details for a workflow:
```

Comment:
The "when they are available" feels extraneous here, I wouldn't think I could view details about something unavailable

Maybe,
```suggestion
Once a workflow is imported you can view the details such as to verify the payload. To view the details for a workflow:
```